### PR TITLE
Makes strange reagent require a shock to activate

### PR DESCRIPTION
**Pull request by Skeletonman0**
[QoL]
## About the PR 
This PR makes Strange Reagent require a shock that gives the "defibbed" status effect with at minimum 1 unit of the chemical in the body to cause its effects.

Methods i have tested of reviving people include:
Batons, Electric Chairs, Grilles, Defibrillators, Arcfiends, and Buildmode arcflashes

## Why's this needed? 
It seems nice, possibly could allow people to pretend to be Dr. Frankenstein.
also blame yass
image

## Changelog

```changelog
(u)Skeletonman0
(*)Made Strange Reagent require an electric shock to restart people's hearts.
```

### CODE REVIEW

**Comment by LadyBirb** [comment]
This would mean you would no longer be able to use it with injector belts or vapomatics to self-revive. If this is a good thing or not is up for debate.

File: code/modules/chemistry/Reagents-Misc.dm [hunk 1]
```diff
@@ -521,17 +521,24 @@ datum
 			depletion_rate = 0.2
 			value = 28 // 3 3 22
 			viscosity = 0.5
+			on_add()
+				..()
+				if(ismob(src.holder?.my_atom))
+					RegisterSignal(holder.my_atom, COMSIG_MOB_SHOCKED_DEFIB, .proc/revive)
```
Comment (ZeWaka) [inline code/modules/chemistry/Reagents-Misc.dm:527]: feel like tasers should work

Comment (ZeWaka) [reply code/modules/chemistry/Reagents-Misc.dm:527]: or any electric shock

**Comment by ZeWaka** [comment]
did you forget to push some commits or something? the PR definitely doesn't do what it says it does

**Comment by Skeletonman0** [comment]
i was using the the "defibbed" effect to make it work, as it seemed like an easy way to do it. defibrillators dont apply that effect to corpses.

**Comment by Skeletonman0** [comment]
i am currently working on trying to make defibrillators trigger it

**Comment by Skeletonman0** [comment]
i changed my mind about wanting to do this.

**Comment by Skeletonman0** [comment]
i gave this another try

**Comment by Tarmunora** [comment]
Honestly I like this approach better -- just make defibbing a corpse give a couple seconds of the status effect and we're good to go

File: code/modules/chemistry/Reagents-Misc.dm (hunk 1 above)
Comment (Skeletonman0) [reply code/modules/chemistry/Reagents-Misc.dm:527]: working on it, almost got defibrillators and tasers to work

Comment (Skeletonman0) [reply code/modules/chemistry/Reagents-Misc.dm:527]: yass says they dont want tasers to work on it so

**Comment by Vaskritaya** [comment]
> 
> 
> This would mean you would no longer be able to use it with injector belts or vapomatics to self-revive. If this is a good thing or not is up for debate.

It might still be possible using voltagen, but I lack the familiarity with voltagen to say. That being said, even if it works it would be more fiddly than current.

File: code/mob/living.dm [hunk 2]
```diff
@@ -1920,7 +1920,8 @@ var/global/icon/human_static_base_idiocy_bullshit_crap = icon('icons/mob/human.d
 /mob/living/shock(var/atom/origin, var/wattage, var/zone = "chest", var/stun_multiplier = 1, var/ignore_gloves = 0)
 	if (!wattage)
 		return 0
-
+	if (isdead(src) && !src.bioHolder.HasEffect("resist_electric"))
+		src.setStatus("defibbed", 1.5 SECONDS)
```
Comment (Tarmunora) [inline code/mob/living.dm:1924]: remove `if` and move to right above the shock damage switch

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/mob/living.dm (hunk 2 above)
Comment (Skeletonman0) [reply code/mob/living.dm]: i think i did it

**Pull request merged**